yateto/arch.py

```diff
@@ -117,17 +117,27 @@ def getArchitectureIdentifiedBy(ident):
 
   # NOTE: ibxsmm currently supports prefetch only for KNL kernels
   arch = {
-    'noarch': Architecture(name, precision, 16, False),
-    'wsm': Architecture(name, precision, 16, False),
-    'snb': Architecture(name, precision, 32, False),
-    'hsw': Architecture(name, precision, 32, False),
+    'noarch': Architecture(name, precision, 64, False),
+    'wsm': Architecture(name, precision, 64, False),
+    'snb': Architecture(name, precision, 64, False),
+    'hsw': Architecture(name, precision, 64, False),
     'skx': Architecture(name, precision, 64, True),
     'knc': Architecture(name, precision, 64, False),
     'knl': Architecture(name, precision, 64, True),
-    'rome': Architecture(name, precision, 32, False),
-    'thunderx2t99': Architecture(name, precision, 16, False),
+    'naples': Architecture(name, precision, 64, False),
+    'rome': Architecture(name, precision, 64, False),
+    'milan': Architecture(name, precision, 64, False),
+    'bergamo': Architecture(name, precision, 64, False),
+    'thunderx2t99': Architecture(name, precision, 64, False),
     'a64fx': Architecture(name, precision, 256, True),
-    'power9': Architecture(name, precision, 16, False)
+    'neon': Architecture(name, precision, 64, False),
+    'apple-m1': Architecture(name, precision, 128, False),
+    'sve128': Architecture(name, precision, 64, False),
+    'sve256': Architecture(name, precision, 64, False),
+    'sve512': Architecture(name, precision, 64, False),
+    'sve1024': Architecture(name, precision, 128, False),
+    'sve2048': Architecture(name, precision, 256, False),
+    'power9': Architecture(name, precision, 64, False)
```

Comment:
Don't change the alignment for any of the existing archs!

The alignment number does not only affect the memory alignment but also the number of hardware flops done in the following way: The alignment is typically coupled to the vector width (e.g. AVX512 => 64 byte vectors), so an optimization trick is to fill up zeros such that you always have full vector instructions (particularly useful if efficient masking is missing, but also useful to have aligned loads everywhere). So, e.g. if you have a matrix with 35 rows (double precision) and an vector width of 64 bytes then you can fit 8 doubles in one vector, therefore yateto fills up with zeros to 40 rows, such that we have 40/8 = 5 full vector instructions per column.
Now if you change the alignment of snb or hsw, e.g., to 64 you also fill up to 40 rows and need 10 vector instructions per column, but you would need only 9 if you only fill up to 36 rows.
=> More flops, no gain.

Reply:
Ok, I guess it may be time to split up the `alignment` into a vector size and a cache line size then (that'll also work towards implementing #60 ).

Reply:
The only difference that would make is in the alignment of the temporary stack arrays. No one prevents you from over-aligning input data outside of yateto. I'm not sure whether over-aligning temporaries is a good idea, so make sure to benchmark.

Reply:
Then IMO the name alignment is incorrect and it should be called padding?
If this attribute only changes the padding, then A64FX should also have 512bit padding. (I haven't reported the performance numbers for this architecture publicly,so it doesn't matter)

Reply:
You can change the name to whatever you like, just don't change the values :-D

Reply:
EDIT: Just benchmarked it, please change A64FX to 512bit padding as well. Pretty sure I merged this incorrectly and always used this padding...

Reply:
The old Fugaku patch (https://github.com/SeisSol/SeisSol/pull/755 ) used 64 byte / 512 bit as well (in SeisSol CMake).

... So is there still a need to differentiate between alignment and vector length? Or can we keep everything as it is?

Reply:
IMO we should differentiate between vector length alignment and cache line alignment at some point. There are different reasons for both alignments. I.e. you want to align to cache line size to avoid false sharing in some cases, to vector instruction length for the kernels, etc

> The old Fugaku patch (https://github.com/SeisSol/SeisSol/pull/755 ) used 64 byte / 512 bit as well (in SeisSol CMake).

Yup. Right now, I'm running with ALIGNMENT=256 and yateto alignment of 64. Not sure whether that's the best choice but it's reasonable. I haven't benchmarked on this type of machine for a long time, as we do not currently have access to any reasonable sized machine with this arch :(

Reply:
Ok, I've reset everything to the vector length.